routes/actions.js

```diff
@@ -201,6 +201,15 @@ router.get('/new/user', function(req, res) {
 });
 
 router.get('/register', function(req, res) {
+  if (req.query.signupLanguage && languages.isValid(req.query.signupLanguage)){
```

Comment:
We should add `&& !req.user` to this check so a malicious user can't switch someone else's language by sending them to a signupLanguage register link. This will simply suppress the feature if the user is already logged in.